compiler/rustc_codegen_llvm/src/va_arg.rs

```diff
@@ -186,8 +186,8 @@ pub(super) fn emit_va_arg(
         ("aarch64", true) => {
             emit_ptr_va_arg(bx, addr, target_ty, false, Align::from_bytes(8).unwrap(), false)
         }
-        // iOS AArch64
-        ("aarch64", _) if target.target_os == "ios" => {
+        // macOS/iOS AArch64
+        ("aarch64", _) if target.target_os == "macos" || target.target_os == "ios" => {
```

Comment:
AFAIK in both the `target_os == "macos"` and in the `target_os == "ios"` case `target.options.is_like_osx == true`. I wonder if it would be simpler to have `("aarch64", _) if target.options.is_like_osx` instead of checking on the string of the OS. Is there a case where `target.options.is_like_osx == true` and the target arch is `"aarch64"`, but the va_list isn't a `char*`? I can't think of one, but I may be missing one.

Reply:
Updated!